tilt-provider.json

```diff
@@ -0,0 +1,9 @@
+{
+    "name": "packet",
+    "config": {
+        "image": "packethost/cluster-api-provider-packet",
+        "live_reload_deps": [
+            "main.go", "go.mod", "go.sum", "api", "controllers", "pkg"
+        ]
+    }
+}
```

Comment:
@gianarb This is the main bit that is needed for Tilt support, however I quickly ran into issues because the Tilt utilities in the CAPI repo assume that the provider components can be built by running kustomize against the /config directory.

The rest of the changes were to facilitate running kustomize against the /config directory and align the manifest sources and generation more with other providers.